samcli/hook_packages/terraform/hooks/prepare/hook.py

```diff
@@ -813,6 +818,22 @@ def _generate_makefile(
         makefile.writelines(makefile_rules)
 
 
+def _generate_backend_override_file(output_directory_path: str):
+    """
+    Generates an override tf file to use a temporary backend
+
+    Parameters
+    ----------
+    output_directory_path: str
+        the output directory path to write the generated makefile
+    """
+    statefile_filename = f"{uuid.uuid4()}.tfstate"
+    override_content = "terraform {\n" '  backend "local" {\n' f'    path = "./{statefile_filename}"\n' "  }\n" "}\n"
```

Comment:
Do the file contents matter here and will it be read/executed on WIndows? If so should we use `os.linesep`?

Reply:
We will have another task to address the issues on Windows. Potentially we will generate make rules differently based on OS.